superset-frontend/plugins/plugin-chart-echarts/src/components/Echart.tsx

```diff
@@ -27,8 +27,11 @@ import {
   Ref,
 } from 'react';
 
+import { useSelector } from 'react-redux';
+import { ExplorePageState } from 'src/explore/types';
```

Comment:
@jpchev I think the issue is actually here, because it introduces a dependency between the echart plugin and the superset-frontend code, and so the typescript compiler (tsc) doesn't know how to deal with it.

You can work around it by removing this import:
```suggestion
```

And defining that interface inline below (under the rest of the imports):

```typescript
// Define this interface here to avoid creating a dependency back to superset-frontend, 
// and to appease the compiler.
// ref: superset-frontend/src/explore/types
interface ExplorePageState {
  common: {
    locale: string;
  };
};
```